src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs

```diff
@@ -97,9 +88,31 @@ public CSharpCompilationOptions CompilationOptions
                 {
                     throw new ArgumentNullException(nameof(value));
                 }
+
                 _compilationOptions = value;
             }
         }
 
+        /// <summary>
+        /// Gets the composite result of <see cref="FileProviders"/>.
+        /// </summary>
+        /// <returns>The <see cref="IFileProvider"/> instance if exactly one instance is registered, otherwise
+        /// a <see cref="CompositeFileProvider"/> that contains all registered instances.</returns>
+        public IFileProvider GetCompositeFileProvider()
```

Comment:
Should we just move this to a facade service? I don't think this is horrid, just kinda ugly to put on options (which the user will see).